patches/tModLoader/Terraria/ModLoader/LegacyPlayerDrawLayer.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using Terraria.DataStructures;
+
+namespace Terraria.ModLoader
+{
+	/// <summary> This class represents a DrawLayer for the player, and uses PlayerDrawInfo as its InfoType. Drawing should be done by adding Terraria.DataStructures.DrawData objects to Main.playerDrawData. </summary>
+	[Autoload(false)]
+	public sealed class LegacyPlayerDrawLayer : PlayerDrawLayer
```

Comment:
should this be private? I see it's used by `ModLoaderMod` but should it be? I get that it might be useful to provide a lambda for the draw func, maybe there's a way to roll the inherit vs functional constructor versions together? or at least rename this to not include `Legacy`.

Passing `Mod` to the constructor also feels wrong, since layers are allocated in the dict via what mod the hook adding them belongs to. I know we've used ModType in this inheritance chain, but did we need to? Is anyone going to enumerate layers and want to know what mod they came from/what their name is and not have that info? For debugging it'll be in the stack trace